common/src/Model/Brush.h

```diff
@@ -67,7 +71,17 @@ namespace TrenchBroom {
             mutable bool m_transparent;
             mutable std::unique_ptr<Renderer::BrushRendererBrushCache> m_brushRendererBrushCache; // unique_ptr for breaking header dependencies
         public:
-            Brush(const vm::bbox3& worldBounds, const std::vector<BrushFace*>& faces);
+            static kdl::result<std::unique_ptr<Brush>, GeometryException> create(const vm::bbox3& worldBounds, const std::vector<BrushFace*>& faces);
+        private:
+            Brush();
+            kdl::result<void, GeometryException> initialize(const vm::bbox3& worldBounds, const std::vector<BrushFace*>& faces);
+        public:
+            Brush(const Brush& other);
+            Brush(Brush&& other) noexcept;
```

Comment:
I'm not sure about having swap/copy/move constructors on a Node subclass, since they need to ignore the node tree membership. Maybe this would be addressed by the splitting of Brush and BrushNode?

Reply:
This is exactly the reason why I'm splitting them up, yes. I'll redo this PR when I'm done with the splitting (and that PR is merged).